lambda/src/main/scala/pricemigrationengine/migrations/Newspaper2024Migration/Newspaper2024MigrationAmendment.scala

```diff
@@ -0,0 +1,89 @@
+package pricemigrationengine.migrations
+import pricemigrationengine.migrations.Newspaper2024MigrationEstimation.SubscriptionData2024
+import pricemigrationengine.migrations.Newspaper2024MigrationStaticData._
+import pricemigrationengine.model._
+
+import java.time.LocalDate
+
+object Newspaper2024MigrationAmendment {
+
+  def subscriptionToNewChargeDistribution2024(subscription: ZuoraSubscription): Option[ChargeDistribution2024] = {
+    val priceCorrectionFactor = Newspaper2024MigrationPriceCapping.priceCorrectionFactor(subscription)
+    for {
+      data2024 <- Newspaper2024MigrationEstimation
+        .subscriptionToSubscriptionData2024(subscription)
+        .toOption
+      priceDistribution <- Newspaper2024MigrationStaticData.priceDistributionLookup(
+        data2024.productName,
+        data2024.billingPeriod,
+        data2024.ratePlanName
+      )
+    } yield chargeDistributionMultiplier(priceDistribution, priceCorrectionFactor)
+  }
+
+  def chargeDistributionToChargeOverrides(
+      distribution: ChargeDistribution2024,
+      billingPeriod: String
+  ): List[ChargeOverride] = {
+    List(
+      distribution.monday,
+      distribution.tuesday,
+      distribution.wednesday,
+      distribution.thursday,
+      distribution.friday,
+      distribution.saturday,
+      distribution.sunday,
+      distribution.digitalPack,
+    ).flatten.map { individualCharge =>
+      ChargeOverride(
+        productRatePlanChargeId = individualCharge.chargeId,
+        billingPeriod = billingPeriod,
+        price = individualCharge.Price
+      )
+    }
+  }
+
+  def subscriptionToZuoraSubscriptionUpdate(
+      subscription: ZuoraSubscription,
+      effectiveDate: LocalDate,
+  ): Either[AmendmentDataFailure, ZuoraSubscriptionUpdate] = {
+
+    def transform1[T](option: Option[T]): Either[AmendmentDataFailure, T] = {
+      option match {
+        case None                 => Left(AmendmentDataFailure("error"))
+        case Some(ratePlanCharge) => Right(ratePlanCharge)
+      }
+    }
+
+    def transform2[T](data: Either[String, T]): Either[AmendmentDataFailure, T] = {
+      data match {
+        case Left(string) => Left(AmendmentDataFailure(string))
+        case Right(t)     => Right(t)
+      }
```

Comment:
```suggestion
      data.left.map(AmendmentDataFailure)
```

Reply:
Done. So much better, thank you for that Kelvin. Now the two `transform2` functions are gone. Also updating the PR description